deluge/ui/gtkui/connectionmanager.py

```diff
@@ -233,6 +234,23 @@ def add_host(self, host, port, username='', password=''):
         # Update the status of the hosts
         self.__update_list()
 
+    def on_entry_host_paste_clipboard(self, widget):
+        paste = gtk.clipboard_get().wait_for_text()
+        log.debug('on_entry_proxy_host_paste-clipboard: got paste: %s', paste)
+        if '//' not in paste:
+            parsed = urlparse('//' + paste)
+        else:
+            parsed = urlparse(paste)
+        if parsed.hostname is not None:
+            widget.set_text(parsed.hostname)
+            widget.emit_stop_by_name('paste-clipboard')
+        if parsed.port is not None:
+            self.builder.get_object('spinbutton_port').set_value(parsed.port)
+        if parsed.username is not None:
+            self.builder.get_object('entry_username').set_text(parsed.username)
+        if parsed.password is not None:
+            self.builder.get_object('entry_password').set_text(parsed.password)
```

Comment:
I don't understand what you mean by this? Would you rather have `TypeError: Gtk.Entry.set_text() argument 1 must be string, not None` each time one of these is None?